Makefile

```diff
@@ -1,17 +1,33 @@
+.PHONY: all fmt check-fmt lint lint-fix codespell all
+
+DESTDIR ?=
 
 ALL_SRC_FILES = 'find . -name "*.[ch]" ! -path "./src/libhirte/ini/ini.[ch]" ! -path "./src/libhirte/hashmap/*" ! -path "./src/*/test/**" ! -path "./builddir/**" -print0'
 
+all:
+	meson setup builddir
+	meson compile -C builddir
+
+test: all
+	meson setup builddir
+	meson compile -C builddir
+	meson test -C builddir
+
+install: all
+	meson install -C builddir --destdir "$(DESTDIR)"
+
```

Comment:
What do you think of a `clean` target? For example, like this
```bash
rm -rf builddir
meson setup builddir
```

Reply:
Yes definitely

Reply:
Ninja has a clean target -- it cleans all built outputs, but doesn't remove the build directory entirely, so you still have a ninja file. Adding a Makefile target that purges the entire build directory seems confusingly different from what meson/ninja does -- and it's also different from what other build systems do.

e.g. autotools distinguishes between `make clean` (removes built outputs) and `make distclean` (removes generated builddir, Makefiles, etc.)